src/Core/functions.php

```diff
@@ -38,7 +38,7 @@
      * @return mixed Wrapped text.
      * @link https://book.cakephp.org/4/en/core-libraries/global-constants-and-functions.html#h
      */
-    function h($text, bool $double = true, ?string $charset = null)
+    function h(mixed $text, bool $double = true, ?string $charset = null): mixed
```

Comment:
This should be `string`.

Reply:
We can make this change in 4.3 I think.

Reply:
I tried this in 4.3, but not comfortable casting the scalars to string. Would rather error if a bool is passed in.